lib/fileutils.rb

```diff
@@ -358,25 +358,54 @@ def rmdir(list, parents: nil, noop: nil, verbose: nil)
   module_function :rmdir
 
   #
-  # :call-seq:
-  #   FileUtils.ln(target, link, force: nil, noop: nil, verbose: nil)
-  #   FileUtils.ln(target,  dir, force: nil, noop: nil, verbose: nil)
-  #   FileUtils.ln(targets, dir, force: nil, noop: nil, verbose: nil)
+  # When +src+ is the path to an existing file or directory
+  # and +dest+ is the path to a non-existent file,
+  # creates a hard link at +dest+ pointing to +src+; returns zero:
+  #
+  #   Dir.children('tmp0/')                    # => ["t.txt"]
+  #   Dir.children('tmp1/')                    # => []
+  #   FileUtils.ln('tmp0/t.txt', 'tmp1/t.lnk') # => 0
+  #   Dir.children('tmp1/') # => ["t.lnk"]
+  #
+  # When +src+ is the path to an existing file or directory
```

Comment:
I don't think `src` can be a directory, it raises an error for me:

```ruby
irb(main):005:0> Dir.children('dir1')
=> ["test.txt"]
irb(main):006:0> Dir.children('dir2')
=> []
irb(main):007:0> FileUtils.ln('dir1', 'dir2')
/Users/peter/.rubies/ruby-3.1.2/lib/ruby/3.1.0/fileutils.rb:301:in `link': Operation not permitted @ rb_file_s_link - (dir1, dir2/dir1) (Errno::EPERM)
```

Reply:
Fixed.